File: backend/app/Models/Reserva.php
<?php


namespace App\Models;


use Illuminate\Database\Eloquent\Model;

class Reserva extends Model {
    public $timestamps = false;
    protected $casts = [
        'acompanantes' => 'array',
        'matriculas' => 'array',
        'incidencias' => 'array'
    ];
    protected $fillable = [
        'id_persona','dni_persona','fecha_entrada','fecha_salida',
        'localizador_reserva','num_personas','acompanantes','num_vehiculos',
        'num_casetas','num_bus','num_caravanas', 'matriculas','checkin','fecha_checkin',
        'incidencia','incidencias','estado','id_zona'
        ];
    protected $table="reservas";
}

File: frontend/app/src/main/java/com/example/frontend/controller/models/Reserva.kt
package com.example.frontend.controller.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.squareup.moshi.Json
import org.json.JSONObject


@Entity(tableName = "reservas")
data class Reserva(@PrimaryKey(autoGenerate = true)
                   val id: Int,
                   val id_persona: Int,
                   val dni_persona: String,
                   val fecha_entrada: String,
                   val fecha_salida: String,
                   val localizador_reserva: String,
                   val num_personas: Int,
                   val acompanantes: String,
                   val num_vehiculos: Int,
                   val num_casetas: Int,
                   val num_bus: Int,
                   val num_caravanas: Int,
                   val matriculas: String,
                   val checkin: String,
                   val fecha_checkin: String,
                   val incidencia: String,
                   val incidencias: String,
                   val estado: String,
                   val id_zona: Int)